File: DAL/src/main/java/com/binwell/dal/DataServices.java
package com.binwell.dal;

import com.binwell.dal.api.BusStationApi;
import com.binwell.dal.api.NewRouteApi;
import com.binwell.dal.api.ObjectApi;
import com.binwell.dal.api.RouteApi;

public class DataServices {

    public static BusStationApi BusStationService;
    public static NewRouteApi NewRouteService;
    public static ObjectApi ObjectService;
    public static RouteApi RouteService;

    public static void init() {
        ApiClient client = new ApiClient();
        client.createDefaultAdapter();
        BusStationService = client.createService(BusStationApi.class);
        NewRouteService = client.createService(NewRouteApi.class);
        ObjectService = client.createService(ObjectApi.class);
        RouteService = client.createService(RouteApi.class);
    }
}

File: DAL/src/main/java/ru/boomik/vrnbus/dal/DataServices.kt
package ru.boomik.vrnbus.dal

import ru.boomik.vrnbus.dal.api.ICoddApi
import ru.boomik.vrnbus.dal.remote.services.CoddDataService
import ru.boomik.vrnbus.dal.remote.services.CoddPersistentDataService
import kotlin.reflect.KFunction1

object DataServices {
    lateinit var CoddDataService: CoddDataService
    lateinit var CoddPersistentDataService: CoddPersistentDataService

    fun init(cachePath: String, logFunc: KFunction1<String, Unit>) {
        val client = ApiClient()

        val service = client.createService(ICoddApi::class.java)
        CoddDataService = CoddDataService(service)
        CoddPersistentDataService = CoddPersistentDataService(service, cachePath, 3 * 24 * 60 * 60 * 1000/*3 days*/,logFunc)
    }
}